worker/src/scripts/upsertDefaultModelPrices.ts

```diff
@@ -0,0 +1,192 @@
+          where: {
+            projectId: null,
+          },
+          select: {
+            id: true,
+            updatedAt: true,
+          },
+        })
+      ).map((model) => [model.id, model.updatedAt])
+    );
+
+    // Upsert in batches
+    const batchSize = 10;
+    const numBatches = Math.ceil(parsedDefaultModelPrices.length / batchSize);
+
+    for (let i = 0; i < numBatches; i++) {
+      logger.debug(`Processing batch ${i + 1} of ${numBatches}...`);
+
+      const batch = parsedDefaultModelPrices.slice(
+        i * batchSize,
+        (i + 1) * batchSize
+      );
+
+      const promises = [];
+
+      for (const defaultModelPrice of batch) {
+        const existingModelUpdateDate = existingModelUpdateDates.get(
+          defaultModelPrice.id
+        );
+
+        if (
+          !force &&
+          existingModelUpdateDate &&
+          existingModelUpdateDate > defaultModelPrice.updated_at
+        ) {
+          logger.error(
+            `Model drift detected for default model ${defaultModelPrice.model_name} (${defaultModelPrice.id}). updatedAt ${existingModelUpdateDate} after ${defaultModelPrice.updated_at}.`
+          );
+          continue;
+        }
+
+        if (
+          !force &&
+          existingModelUpdateDate &&
+          existingModelUpdateDate.getTime() ==
+            defaultModelPrice.updated_at.getTime()
+        ) {
+          logger.debug(
+            `Default model ${defaultModelPrice.model_name} (${defaultModelPrice.id}) already up to date. Skipping.`
+          );
+          continue;
+        }
+
+        // Upsert model and prices in a transaction
+        promises.push(
+          prisma
+            .$transaction(async (tx) => {
+              await tx.model.upsert({
+                where: {
+                  projectId: null,
+                  id: defaultModelPrice.id,
+                },
+                update: {
+                  modelName: defaultModelPrice.model_name,
+                  matchPattern: defaultModelPrice.match_pattern,
+                  updatedAt: defaultModelPrice.updated_at,
+                  tokenizerConfig:
+                    defaultModelPrice.tokenizer_config ?? undefined,
+                  tokenizerId: defaultModelPrice.tokenizer_id,
+                },
+                create: {
+                  projectId: null,
+                  id: defaultModelPrice.id,
+                  modelName: defaultModelPrice.model_name,
+                  matchPattern: defaultModelPrice.match_pattern,
+                  tokenizerConfig:
+                    defaultModelPrice.tokenizer_config ?? undefined,
+                  tokenizerId: defaultModelPrice.tokenizer_id,
+                  createdAt: defaultModelPrice.created_at,
+                  updatedAt: defaultModelPrice.updated_at,
+                },
+              });
+
+              for (const [usageType, price] of Object.entries(
+                defaultModelPrice.prices
+              )) {
```

Comment:
Here, a Promise.all might also help with a shorter transaction duration compared to the for of loop.

Reply:
Yes agree, again did not push for heavy optimization due to only few prices per model